Source/WebCore/editing/FrameSelection.cpp

```diff
@@ -1848,11 +1848,12 @@ Color CaretBase::computeCaretColor(const RenderStyle& elementStyle, const Node*
     // On iOS, we want to fall back to the tintColor, and only override if CSS has explicitly specified a custom color.
 #if PLATFORM(IOS_FAMILY) && !PLATFORM(MACCATALYST)
     UNUSED_PARAM(node);
-    return elementStyle.caretColor();
+    return elementStyle.colorResolvingCurrentColor(elementStyle.caretColor());
 #else
     RefPtr parentElement = node ? node->parentElement() : nullptr;
     auto* parentStyle = parentElement && parentElement->renderer() ? &parentElement->renderer()->style() : nullptr;
-    if (elementStyle.hasAutoCaretColor() && parentStyle) {
+    // CSS value "auto" is treated as an invalid color.
+    if (!elementStyle.colorResolvingCurrentColor(elementStyle.caretColor()).isValid() && parentStyle) {
```

Comment:
This seems like a more verbose way to say the same thing, if the check for `isValid()` is needed now, the comment probably needs to be changed. Otherwise, we should do this change once we actually need it.

Reply:
Yes this change is not needed since the rebasing.